### DISCUSSION TITLE
python predict_system.py 报错： error: unrecognized arguments: --table_model_dir

### DISCUSSION THREAD

**Discussion by lygyds** [PaddlePaddle/PaddleOCR#14172] [Q&A] [answered]
环境：paddlepaddle2.6.1  paddleocr2.9.1

运行：python C:\Users\13100\.conda\envs\pp\Lib\site-packages\paddleocr\tools\infer\predict_system.py --image_dir=ad1.jpg --det_model_dir=inference\en_PP-OCRv3_det_infer --rec_model_dir=inference\en_PP-OCRv3_rec_infer --rec_char_dict_path=en_dict.txt --table_model_dir=inference\en_ppstructure_mobile_v2.0_SLANet_infer --table_char_dict_path=table_structure_dict.txt --layout_model_dir=inference\picodet_lcnet_x1_0_fgd_layout_infer --layout_dict_path=layout_publaynet_dict.txt --vis_font_path=arial.ttf --recovery=True --output=..\output\ --return_word_box=True
 报错：
predict_system.py: error: unrecognized arguments: --table_model_dir=inference\en_ppstructure_mobile_v2.0_SLANet_infer --table_char_dict_path=table_structure_dict.txt --layout_model_dir=inference\picodet_lcnet_x1_0_fgd_layout_infer --layout_dict_path=layout_publaynet_dict.txt --recovery=True --output=..\output\

**Comment by UserWangZz** [comment, marked as answer]
参考文档吧，命令输入错了，可能和windows输入命令python将后面参数解析成一个argu了

**Comment by lygyds** [comment]
[图片1](https://github.com/user-attachments/assets/00d92a08-b3f2-40fe-9fa9-1b1b957f6626)

--table_char_dict_path，在predict_system.py中找不到。

**Comment by lygyds** [comment]
参考的是官网代码：返回识别位置，
https://paddlepaddle.github.io/PaddleOCR/latest/ppstructure/blog/return_word_pos.html#_3

--det_model_dir=...等等，其它参数都没有问题。

**Comment by GreatV** [comment]
无法复现

```text
[2024/11/07 01:42:16] ppocr WARNING: The first GPU is used for inference by default, GPU ID: 0
[2024/11/07 01:42:17] ppocr WARNING: The first GPU is used for inference by default, GPU ID: 0
[2024/11/07 01:42:17] ppocr WARNING: The first GPU is used for inference by default, GPU ID: 0
[2024/11/07 01:42:18] ppocr WARNING: The first GPU is used for inference by default, GPU ID: 0
[2024/11/07 01:42:18] ppocr WARNING: The first GPU is used for inference by default, GPU ID: 0
[2024/11/07 01:42:19] ppocr INFO: [0/1] ../docs/ppstructure/images/table_1.png
[2024/11/07 01:42:19] ppocr DEBUG: dt_boxes num : 214, elapsed : 0.0809030532836914
[2024/11/07 01:42:20] ppocr DEBUG: rec_res num  : 214, elapsed : 0.3577108383178711
[2024/11/07 01:42:20] ppocr DEBUG: dt_boxes num : 113, elapse : 0.022031784057617188
[2024/11/07 01:42:20] ppocr DEBUG: rec_res num  : 113, elapse : 0.13351964950561523
[2024/11/07 01:42:20] ppocr DEBUG: dt_boxes num : 80, elapse : 0.0184018611907959
[2024/11/07 01:42:20] ppocr DEBUG: rec_res num  : 80, elapse : 0.0989389419555664
[2024/11/07 01:42:21] ppocr INFO: result save to ../output/structure/table_1/show_0.jpg
[2024/11/07 01:42:21] ppocr INFO: docx save to ../output/structure/table_1_ocr.docx
[2024/11/07 01:42:21] ppocr INFO: Predict time : 1.315s
```

**Comment by lygyds** [comment]
具体操作如下
(pp) G:\PaddleOCR\ppstructure>python C:\Users\13100\.conda\envs\pp\Lib\site-packages\paddleocr\tools\infer\predict_system.py --table_model_dir=inference\en_ppstructure_mobile_v2.0_SLANet_infer
usage: predict_system.py [-h] [--use_gpu USE_GPU] [--use_xpu USE_XPU] [--use_npu USE_NPU] [--use_mlu USE_MLU]
                         [--ir_optim IR_OPTIM] [--use_tensorrt USE_TENSORRT] [--min_subgraph_size MIN_SUBGRAPH_SIZE]
                         [--precision PRECISION] [--gpu_mem GPU_MEM] [--gpu_id GPU_ID] [--image_dir IMAGE_DIR]
                         [--page_num PAGE_NUM] [--det_algorithm DET_ALGORITHM] [--det_model_dir DET_MODEL_DIR]
                         [--det_limit_side_len DET_LIMIT_SIDE_LEN] [--det_limit_type DET_LIMIT_TYPE]
                         [--det_box_type DET_BOX_TYPE] [--det_db_thresh DET_DB_THRESH]
                         [--det_db_box_thresh DET_DB_BOX_THRESH] [--det_db_unclip_ratio DET_DB_UNCLIP_RATIO]
                         [--max_batch_size MAX_BATCH_SIZE] [--use_dilation USE_DILATION]
                         [--det_db_score_mode DET_DB_SCORE_MODE] [--det_east_score_thresh DET_EAST_SCORE_THRESH]
                         [--det_east_cover_thresh DET_EAST_COVER_THRESH] [--det_east_nms_thresh DET_EAST_NMS_THRESH]
                         [--det_sast_score_thresh DET_SAST_SCORE_THRESH] [--det_sast_nms_thresh DET_SAST_NMS_THRESH]
                         [--det_pse_thresh DET_PSE_THRESH] [--det_pse_box_thresh DET_PSE_BOX_THRESH]
                         [--det_pse_min_area DET_PSE_MIN_AREA] [--det_pse_scale DET_PSE_SCALE] [--scales SCALES]
                         [--alpha ALPHA] [--beta BETA] [--fourier_degree FOURIER_DEGREE]
                         [--rec_algorithm REC_ALGORITHM] [--rec_model_dir REC_MODEL_DIR]
                         [--rec_image_inverse REC_IMAGE_INVERSE] [--rec_image_shape REC_IMAGE_SHAPE]
                         [--rec_batch_num REC_BATCH_NUM] [--max_text_length MAX_TEXT_LENGTH]
                         [--rec_char_dict_path REC_CHAR_DICT_PATH] [--use_space_char USE_SPACE_CHAR]
                         [--vis_font_path VIS_FONT_PATH] [--drop_score DROP_SCORE] [--e2e_algorithm E2E_ALGORITHM]
                         [--e2e_model_dir E2E_MODEL_DIR] [--e2e_limit_side_len E2E_LIMIT_SIDE_LEN]
                         [--e2e_limit_type E2E_LIMIT_TYPE] [--e2e_pgnet_score_thresh E2E_PGNET_SCORE_THRESH]
                         [--e2e_char_dict_path E2E_CHAR_DICT_PATH] [--e2e_pgnet_valid_set E2E_PGNET_VALID_SET]
                         [--e2e_pgnet_mode E2E_PGNET_MODE] [--use_angle_cls USE_ANGLE_CLS]
                         [--cls_model_dir CLS_MODEL_DIR] [--cls_image_shape CLS_IMAGE_SHAPE] [--label_list LABEL_LIST]
                         [--cls_batch_num CLS_BATCH_NUM] [--cls_thresh CLS_THRESH] [--enable_mkldnn ENABLE_MKLDNN]
                         [--cpu_threads CPU_THREADS] [--use_pdserving USE_PDSERVING] [--warmup WARMUP]
                         [--sr_model_dir SR_MODEL_DIR] [--sr_image_shape SR_IMAGE_SHAPE] [--sr_batch_num SR_BATCH_NUM]
                         [--draw_img_save_dir DRAW_IMG_SAVE_DIR] [--save_crop_res SAVE_CROP_RES]
                         [--crop_res_save_dir CROP_RES_SAVE_DIR] [--use_mp USE_MP]
                         [--total_process_num TOTAL_PROCESS_NUM] [--process_id PROCESS_ID] [--benchmark BENCHMARK]
                         [--save_log_path SAVE_LOG_PATH] [--show_log SHOW_LOG] [--use_onnx USE_ONNX]
                         [--return_word_box RETURN_WORD_BOX]
predict_system.py: error: unrecognized arguments: --table_model_dir=inference\en_ppstructure_mobile_v2.0_SLANet_infer

**Comment by lygyds** [comment]
这是完整的操作
(pp) G:\PaddleOCR\ppstructure>python C:\Users\13100\.conda\envs\pp\Lib\site-packages\paddleocr\tools\infer\predict_system.py --image_dir=ad1.jpg --det_model_dir=inference\en_PP-OCRv3_det_infer --rec_model_dir=inference\en_PP-OCRv3_rec_infer --rec_char_dict_path=en_dict.txt --table_model_dir=inference\en_ppstructure_mobile_v2.0_SLANet_infer --table_char_dict_path=table_structure_dict.txt --layout_model_dir=inference\picodet_lcnet_x1_0_fgd_layout_infer --layout_dict_path=layout_publaynet_dict.txt --vis_font_path=arial.ttf --recovery=True --output=..\output\ --return_word_box=True
usage: predict_system.py [-h] [--use_gpu USE_GPU] [--use_xpu USE_XPU] [--use_npu USE_NPU] [--use_mlu USE_MLU] [--ir_optim IR_OPTIM]
                         [--use_tensorrt USE_TENSORRT] [--min_subgraph_size MIN_SUBGRAPH_SIZE] [--precision PRECISION] [--gpu_mem GPU_MEM]
                         [--gpu_id GPU_ID] [--image_dir IMAGE_DIR] [--page_num PAGE_NUM] [--det_algorithm DET_ALGORITHM] [--det_model_dir DET_MODEL_DIR]
                         [--det_limit_side_len DET_LIMIT_SIDE_LEN] [--det_limit_type DET_LIMIT_TYPE] [--det_box_type DET_BOX_TYPE]
                         [--det_db_thresh DET_DB_THRESH] [--det_db_box_thresh DET_DB_BOX_THRESH] [--det_db_unclip_ratio DET_DB_UNCLIP_RATIO]
                         [--max_batch_size MAX_BATCH_SIZE] [--use_dilation USE_DILATION] [--det_db_score_mode DET_DB_SCORE_MODE]
                         [--det_east_score_thresh DET_EAST_SCORE_THRESH] [--det_east_cover_thresh DET_EAST_COVER_THRESH]
                         [--det_east_nms_thresh DET_EAST_NMS_THRESH] [--det_sast_score_thresh DET_SAST_SCORE_THRESH]
                         [--det_sast_nms_thresh DET_SAST_NMS_THRESH] [--det_pse_thresh DET_PSE_THRESH] [--det_pse_box_thresh DET_PSE_BOX_THRESH]
                         [--det_pse_min_area DET_PSE_MIN_AREA] [--det_pse_scale DET_PSE_SCALE] [--scales SCALES] [--alpha ALPHA] [--beta BETA]
                         [--fourier_degree FOURIER_DEGREE] [--rec_algorithm REC_ALGORITHM] [--rec_model_dir REC_MODEL_DIR]
                         [--rec_image_inverse REC_IMAGE_INVERSE] [--rec_image_shape REC_IMAGE_SHAPE] [--rec_batch_num REC_BATCH_NUM]
                         [--max_text_length MAX_TEXT_LENGTH] [--rec_char_dict_path REC_CHAR_DICT_PATH] [--use_space_char USE_SPACE_CHAR]
                         [--vis_font_path VIS_FONT_PATH] [--drop_score DROP_SCORE] [--e2e_algorithm E2E_ALGORITHM] [--e2e_model_dir E2E_MODEL_DIR]
                         [--e2e_limit_side_len E2E_LIMIT_SIDE_LEN] [--e2e_limit_type E2E_LIMIT_TYPE] [--e2e_pgnet_score_thresh E2E_PGNET_SCORE_THRESH]
                         [--e2e_char_dict_path E2E_CHAR_DICT_PATH] [--e2e_pgnet_valid_set E2E_PGNET_VALID_SET] [--e2e_pgnet_mode E2E_PGNET_MODE]
                         [--use_angle_cls USE_ANGLE_CLS] [--cls_model_dir CLS_MODEL_DIR] [--cls_image_shape CLS_IMAGE_SHAPE] [--label_list LABEL_LIST]
                         [--cls_batch_num CLS_BATCH_NUM] [--cls_thresh CLS_THRESH] [--enable_mkldnn ENABLE_MKLDNN] [--cpu_threads CPU_THREADS]
                         [--use_pdserving USE_PDSERVING] [--warmup WARMUP] [--sr_model_dir SR_MODEL_DIR] [--sr_image_shape SR_IMAGE_SHAPE]
                         [--sr_batch_num SR_BATCH_NUM] [--draw_img_save_dir DRAW_IMG_SAVE_DIR] [--save_crop_res SAVE_CROP_RES]
                         [--crop_res_save_dir CROP_RES_SAVE_DIR] [--use_mp USE_MP] [--total_process_num TOTAL_PROCESS_NUM] [--process_id PROCESS_ID]
                         [--benchmark BENCHMARK] [--save_log_path SAVE_LOG_PATH] [--show_log SHOW_LOG] [--use_onnx USE_ONNX]
                         [--return_word_box RETURN_WORD_BOX]
predict_system.py: error: unrecognized arguments: --table_model_dir=inference\en_ppstructure_mobile_v2.0_SLANet_infer --table_char_dict_path=table_structure_dict.txt --layout_model_dir=inference\picodet_lcnet_x1_0_fgd_layout_infer --layout_dict_path=layout_publaynet_dict.txt --recovery=True --output=..\output\

**Comment by GreatV** [comment]
![image](https://github.com/user-attachments/assets/f2c1bf8d-3e52-45d0-bdd9-5300bdbc8633)

**Comment by lygyds** [comment]
谢谢！
(pp) G:\PaddleOCR\ppstructure>python C:\Users\13100\.conda\envs\pp\Lib\site-packages\paddleocr\tools\infer\predict_system.py --image_dir=ad1.jpg --det_model_dir=inference\en_PP-OCRv3_det_infer --rec_model_dir=inference\en_PP-OCRv3_rec_infer --rec_char_dict_path=en_dict.txt --table_model_dir=inference\en_ppstructure_mobile_v2.0_SLANet_infer --table_char_dict_path=table_structure_dict.txt --layout_model_dir=inference\picodet_lcnet_x1_0_fgd_layout_infer --layout_dict_path=layout_publaynet_dict.txt --vis_font_path=arial.ttf --recovery=True --output=..\output\ --return_word_box=True

  --det_model_dir等参数都正常
--table_model_dir这个参数报错（注：ppstructure\inferenc,等文件路径正常）

**Comment by lygyds** [comment]
可以了，非常感谢！
放在ppstructure这个安装目录下就可以了！
